lib/cloud/awsconfig/awsconfig.go

```diff
@@ -274,6 +289,10 @@ func buildConfigOptions(region string, cred aws.CredentialsProvider, opts *optio
 
 // getBaseConfig returns an AWS config without assuming any roles.
 func getBaseConfig(ctx context.Context, region string, opts *options) (aws.Config, error) {
+	if opts.baseConfig != nil {
+		return loadDefaultConfig(ctx, region, opts.baseConfig.Credentials, opts)
```

Comment:
Is `loadDefaultConfig` necessary?  or if only credentials is taken, should we only pass in `baseConfigCredentials`?